src/seamless_communication/cli/m4t/classification_head/train.py

```diff
@@ -0,0 +1,288 @@
+            "without eval loss improvements"
+        ),
+    )
+    parser.add_argument(
+        "--max_epochs",
+        type=int,
+        default=10,
+        help=("Max number of training epochs"),
+    )
+    parser.add_argument(
+        "--learning_rate",
+        type=float,
+        default=1e-7,
+        help=("Finetuning learning rate"),
+    )
+    parser.add_argument(
+        "--warmup_steps",
+        type=int,
+        default=100,
+        help=("Number of steps with linearly increasing learning rate"),
+    )
+    parser.add_argument(
+        "--device",
+        type=str,
+        default="cuda",
+        help=("Device to fine-tune on. See `torch.device`."),
+    )
+    parser.add_argument(
+        "--num_layers", 
+        type=int, 
+        default=2, 
+        help="The number of layers in the classification head"
+    )
+    return parser
+
+
+def plot_losslog(losslog: list, save_to: Path = None, yscale = 'log'):
+    # TODO: Make this look good
+    plt.plot(losslog)
+    plt.yscale(yscale)
+    plt.title('Training Loss')
+    plt.xlabel('Batch')
+    plt.ylabel('Loss')
+    if save_to:
+        plt.savefig(save_to)
+        plt.clf()
+        with open(save_to.parent / "losslog.log", "wb") as f:
+            f.write(losslog)
+    else:
+        plt.show()
+
+
+def train(head: torch.nn.Module,
+          frozen_model: UnitYModel,
+          dataloader: dataloader.UnitYLanguageIDDataLoader,
+          params: ClassificationHeadTrainParams,
+          label_weights: torch.Tensor = None):
+    
+    logger.info("Start Training Language Head")
+    
+    grad_scaler = torch.cuda.amp.GradScaler()
+    optimizer = AdamW(
+        params=head.parameters(),
+        lr=params.learning_rate,
+        betas=(0.9, 0.98),
+        eps=1e-08,
+        maximize=False,
+        weight_decay=0.0,
+        fused=(params.device.type == "cuda"))
+    lr_scheduler = MyleLR(
+        optimizer=optimizer,
+        num_warmup_steps=params.warmup_steps,
+        start_lr=1e-9)
+
+    losslog = list()
+    try:
+        for epoch in range(params.max_epochs):
+            logger.info(f"Epoch {epoch}")
+            
+            # Run batches through train step
+            for seqs, labels in tqdm(dataloader.get_dataloader(), desc="Training Steps"):
+                assert seqs.src_tokens is not None
+                optimizer.zero_grad()
+                seqs.src_tokens = seqs.src_tokens.to(params.device)
+                labels = labels.to(params.device)
+                
+                with torch.autocast(device_type=params.device.type, dtype=params.float_dtype):
+                    mask = PaddingMask(seqs.src_lengths, seqs.src_tokens.size(1)).to(params.device)
+                    vector, _ = frozen_model.encode(seqs.src_tokens, padding_mask=mask)
+                
+                probs = head(vector)
+                
+                loss = torch.nn.functional.cross_entropy(labels, probs)
```

Comment:
it has to be the opposite order of arguments: proba, labels (see documentation)

Reply:
also add label smoothing (0.1-0.2)

Reply:
Done both